.github/workflows/tests.yml

```diff
@@ -60,60 +60,124 @@ jobs:
+        kind: Cluster
+        apiVersion: kind.x-k8s.io/v1alpha4
+        nodes:
+        - role: control-plane
+        - role: worker
+        - role: worker
+        EOF
 
-        # https://microk8s.io/docs/addons
-        sudo microk8s.enable dns
-    - name: Wait for Kubernetes to be ready
-      timeout-minutes: 15
-      run: |
-        while [[ ! $(sudo microk8s.kubectl cluster-info) ]] ; do
-          echo "kubernetes not ready yet..." >&2
-          sleep 1
-        done
     - name: Install CertManager v${{ matrix.certManager }}
       run: |
         # Try the recet way to install crd or fallback to the old one
         version='${{ matrix.certManager }}'
-        shortVersion="${version%.*}"
-        sudo microk8s.kubectl apply -f "https://github.com/jetstack/cert-manager/releases/download/v${version}/cert-manager.crds.yaml" ||
-        sudo microk8s.kubectl apply -f "https://raw.githubusercontent.com/jetstack/cert-manager/release-${shortVersion}/deploy/manifests/00-crds.yaml"
+        kubectl apply -f "https://github.com/jetstack/cert-manager/releases/download/v${version}/cert-manager.yaml"
+        sleep 5
+        time kubectl -n cert-manager wait --for=condition=Available deployment --all --timeout 300s
+
+    - name: Install Ingress
+      run: |
+        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v0.35.0/deploy/static/provider/baremetal/deploy.yaml
+        sleep 5
+        time kubectl -n ingress-nginx wait --for=condition=Available deployment --all --timeout 300s
+
     - uses: actions/checkout@v2
     - name: go tests
       run: |
-        export KUBECONFIG="$(realpath "$(mktemp kubeconfig.XXXXX)")"
-        export USE_EXISTING_CLUSTER="true"
+        make go-test
 
-        sudo microk8s.config > "${KUBECONFIG}"
+    - name: build harbor-operator
+      run: |
+        make docker-build IMG=harbor-operator:dev_test
+        kind load docker-image harbor-operator:dev_test --name harbor
+
+    - name: install harbor-operator
+      run: |
+        set -ex
+        kubectl create ns harbor-operator-ns
+        cd config/default
+        kustomize edit set image goharbor/harbor-operator=harbor-operator:dev_test
+        kustomize build | kubectl apply -f -
+
+        # make helm-install IMG=harbor-operator:dev_test
+
+        if ! time kubectl -n harbor-operator-ns wait --for=condition=Available deployment --all --timeout 300s; then
+          kubectl get all -n harbor-operator-ns
+          exit 1
+        fi
+
+    - name: install harbor
+      run: |
+        export GITHUB_TOKEN=xxx
+        set -ex
+        make sample
+        for i in $(seq 1 6);do
+          sleep 30
+          echo $i
+          kubectl get all
+        done
+        if ! time kubectl wait --for=condition=Ready pod --all --timeout 600s ;then
+          echo install harbor failed
+          kubectl get all
+
+          for n in $(kubectl get po |grep -v Running|grep -v NAME|awk '{print $1}');do
```

Comment:
@bitsf 

I don't think we need to describe each pod here. We can find the unready pods and print out the log  by calling kubectl logs <POD>

Reply:
it has `grep -v Running`